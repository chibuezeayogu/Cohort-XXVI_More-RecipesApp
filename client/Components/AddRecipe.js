import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Row, Input, Icon, Button, Col, Table} from 'react-materialize';
import axios from 'axios';
import Preloader from './Preloder';
//require('dotenv').config();


const CLOUDINARY_URL = process.env.CLOUDINARY_URL;
const CLOUDINARY_UPLOAD_PRESET = process.env.CLOUDINARY_UPLOAD_PRESET;


class AddRecipe extends Component{

    constructor(props) {
        super(props);
        this.state = {
            file: '',
            imagePreviewUrl: '', 
            ImageURL: '',
            arrIngredients:[],
            arrProcedures:[]
        };
      }

      componentWillMount(){
        // if(!this.props.userData.isAuthenticated){
        //     this.context.router.push('/sign');
        // }
        const token = localStorage.getItem('jwtToken');
        if(!token){
            this.context.router.push('/signin');
        }
    }
    //   componentWillUpdate(nextProps, nextState) {
    //     console.log(nextState.arrIngredients);
    // }
   
      _addIngredients(e){
          e.preventDefault();
          const ingredients = document.getElementById('ingredients').value;
          if(!ingredients){
                alert("Ingredients field is empty");
          }else {

            const newIngredients = this.state.arrIngredients.concat(ingredients);
                this.setState({
                    arrIngredients: newIngredients
                });
            }
      }

     



      _addProcedures(e){
        e.preventDefault();
        const procedures = document.getElementById('procedures').value;
        if(!procedures){
              alert("Ingredients field is empty");
        }else {

          const newProcedures = this.state.arrProcedures.concat(procedures);
              this.setState({
                    arrProcedures: newProcedures
              })
          }
        
          this.ingredient.state.reset();
      }

      _handleImageChange(e) {
        e.preventDefault();
    
        let reader = new FileReader();
        let file = e.target.files[0];
        const formData = new FormData;
        formData.append('file', file);
        formData.append('upload_preset', CLOUDINARY_UPLOAD_PRESET);
        
        // if (!file.type.match('image.*')){

        // }else{

        // }
        
        $.ajax({
            type: "POST",
            url: CLOUDINARY_URL,
            dataType: 'json',
            data: formData,
            contentType:false,
            processData: false,
            success: ((data) => {
                this.setState({ImageURL:data.secure_url});
            })
        });

        reader.onloadend = () => {
          this.setState({
            file: file,
            imagePreviewUrl: reader.result
          });
        };
    
        reader.readAsDataURL(file);
      }
      
    handleOnsubmit(e) {
        e.preventDefault();
        const title = this.title.state.value;
        const description = this.description.state.value;
        const ingredients = this.ingredients.state.value;
        const procedures = this.procedures.state.value;
        const imageURL = this.state.ImageURL;
        this.props.addRecipe(title, description, ingredients, procedures, imageURL);
        this.refs.AddRecipeForm.reset();
        this.setState({
            imagePreviewUrl:''
        });
    }
    render() {
        // let {imagePreviewUrl} = this.state;
        // let $imagePreview = null;
        // if (imagePreviewUrl) {
        //   $imagePreview =(<img  class="responsive-img circle" src={imagePreviewUrl}/>);
        // } else {
        //   $imagePreview = " ";
        // }
       const  { arrIngredients, arrProcedures } = this.state;
        return (
            <div className="container"> 
                <h1 className="center">Add Recipe</h1>
                <hr/>
                <form encType="multipart/form-data" methos="post" className="col l6 m8 s12 offset-l8 offset-m6" onSubmit={(e) => this.handleOnsubmit(e)} ref="AddRecipeForm">
                    <Row>
                        <Input s={12} label="Title"  type='text' ref={(input)=> {this.title=input}}><Icon>short_text</Icon></Input>
                    </Row>
                    <Row s={8}>
                        <Input s={12} label="Description" className='validate' type='text' ref={(input) => {this.description = input}}><Icon>short_text</Icon></Input>
                    </Row>
                    <Row>
                        <Input s={12} label="Ingredients" id="ingredients" type='text' ref={(input) => {this.ingredients = input}}><Icon>list</Icon></Input>
                        <Button waves='light' className='right' onClick={(e)=> {this._addIngredients(e)}}>Add Ingredient</Button>
                    </Row>
                    <Row>
                        <Table className="striped">
                            <tbody>
                                {arrIngredients
                                    .map((ingredient, i) => 
                                        <tr>
                                            <td className="s10">
                                                <input id="ingredients" type='text' value={ingredient}></input>
                                            </td>
                                            <td className="s1 right"><Icon>delete</Icon></td>
                                            <td className="s1 right"><Icon>edit</Icon></td>
                                        </tr>
                                    )
                                }
                            </tbody>
                        </Table>
                    </Row>
                    <Row>
                        <Input s={12} label="Procedures" id="procedures" type='text' ref={(input) => {this.procedures = input}}><Icon>list</Icon></Input>
                        <Button waves='light' className='right' onClick={(e)=> {this._addProcedures(e)}}>Add Procedures</Button>
                    </Row>
                    <Row>
                        <Table className="striped">
                            <tbody>
                                {arrProcedures
                                    .map((procedures, i) => 
                                        <tr>
                                            <td className="s10">{procedures}</td>
                                            <td className="s1 right"><Icon>delete</Icon></td>
                                            <td className="s1 right"><Icon>edit</Icon></td>
                                        </tr>
                                    )
                                }
                            </tbody>
                        </Table>
                    </Row>
                    <Row>
                        <Input s={6} id="recipe"  name="file" type='file' ref={(input) => {this.file = input}}   onChange={(e)=>this._handleImageChange(e)}><Icon>insert_photo</Icon></Input>
                        <Col s={6} ><img className="responsive-img img-wh z-depth-3" src={this.state.imagePreviewUrl && this.state.ImageURL?this.state.imagePreviewUrl:"select image to preview"}/></Col>
                    </Row>
                    <br />
                    <Row>
                        <Button waves='light' className='right'><Icon right>send</Icon>Add</Button>
                    </Row>
                </form>
            </div>
        );
    }
}

AddRecipe.contextTypes = {
    router: React.PropTypes.object
};


AddRecipe.propTypes = {
handleOnSubmit: PropTypes.func.isRequired,
_addIngredients:  PropTypes.func.isRequired,
_addProcedures:  PropTypes.func.isRequired,
_handleImageChange:  PropTypes.func.isRequired,
addRecipe: PropTypes.func.isRequired,
};

export default AddRecipe;