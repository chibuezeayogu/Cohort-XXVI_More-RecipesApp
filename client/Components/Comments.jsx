import React from 'react';
import moment from 'moment';
import PropTypes from 'prop-types';

const Comments = props => (
  <li className="collection-item avatar z-depth-1">
    <img
      src={props.review.User.imageUrl}
      alt="profile"
      className="circle responsive-img"
    />
    <span className="text-title bold">{props.review.User.firstName}</span>
    <p
      style={{ wordWrap: 'break-word' }}
    >
      {props.review.comment}
    </p>
    <div className="secondary-content">
      { moment(new Date(props.review.createdAt)).fromNow()}
    </div>
  </li>
);

Comments.propTypes = {
  review: PropTypes.shape({
    comment: PropTypes.string,
    createdAt: PropTypes.string,
    User: PropTypes.shape({
      firstName: PropTypes.string,
      imageUrl: PropTypes.string
    })
  }).isRequired
};

export default Comments;
