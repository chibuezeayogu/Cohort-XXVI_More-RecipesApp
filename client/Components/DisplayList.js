import React, { Component } from 'react';
import { Link } from 'react-router';

const Displaylist = React.createClass ({
    render(){
        const { recipes } = this.props;
        return(
            <td>
                list ingradients
            </td>
        );
    }
});

export default Displaylist;