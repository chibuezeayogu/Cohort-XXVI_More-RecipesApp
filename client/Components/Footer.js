import React, { Component }from 'react';
import PropTypes from 'prop-types';
import { Footer } from 'react-materialize';


class BottomFooter extends  Component {
    render() {
        return (
            <div>
                <Footer copyrights="&copy 2017 Copyright Text"
                moreLinks={
                    <a className="grey-text text-lighten-4 right" href="#!">Andela Project by Ayogu Chibueze></a>
                }
                
                className='black'
            >
                    
                </Footer>;
            </div>
        );
    }
}

BottomFooter.propTypes = {
    logout: PropTypes.func.isRequired,
    history: PropTypes.func.isRequired,
    push: PropTypes.func.isRequired,
    className: PropTypes.string,
    copyrights: PropTypes.string,
};

export default BottomFooter;