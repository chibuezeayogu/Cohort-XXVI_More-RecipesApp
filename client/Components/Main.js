import React, { Component } from 'react';
import { Row, Input, Icon, Textarea } from 'react-materialize';
import NavBar from  './NavBar';
import BottomFooter from  './Footer';
import './../js/recipes.js';
import './../js/script.js';
import Home from './Home';
import SignIn from './SignIn';
import SignUp from './SignUp';

//import sass css
import '../Style/Style.scss';

class Main extends Component {
    constructor(){
        super();
        
    }

    
    render() {
        return (
            <div className="body-style">
                <header>
                    <NavBar {...this.props} />
                </header>
                <main className="main-style">
                    <div className="container-fluid">
                    {React.cloneElement(this.props.children, this.props)}
                    </div>
                </main>
                <footer>
                    <BottomFooter />
                </footer>
            </div>
        )
    }
}

export default Main;