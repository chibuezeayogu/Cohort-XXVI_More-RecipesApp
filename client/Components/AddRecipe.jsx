require('dotenv').config();

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import axios from 'axios';
import { connect } from 'react-redux';
import { createRecipe } from '../action/actionCreators';
import SmallPreloader from './SmallPreloader.jsx';
import Footer from './Footer/Footer.jsx';
import UserMenu from './Header/UserMenu.jsx';
import { validateAddRecipe } from '../middleware/validateInputs';

const CLOUDINARY_URL = process.env.CLOUDINARY_URL;
const CLOUDINARY_UPLOAD_PRESET = process.env.CLOUDINARY_UPLOAD_PRESET;

/**
 * @description AddRecipe component for user to create a new recipe
 *
 * @class
 *
 * @extends Component
 *
 */
class AddRecipe extends Component {
   /**
   * @description initialize state and binds functiom
   *
   * @constructor
   *
   * @memberOf AddRecipe
   *
   * @returns {void}
   *
   */
  constructor() {
    super();
    this.state = {
      title: '',
      description: '',
      ingredients: '',
      procedures: '',
      imageUrl: '',
      image: {},
      errors: {},
      disabled: false,
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleOnsubmit = this.handleOnsubmit.bind(this);
  }

   /**
   * @description routes to /recipes if user is authenticated
   *
   * @method
   *
   * @memberOf AddRecipe
   *
   * @returns {void}
   *
   */
  componentWillMount() {
    const token = localStorage.getItem('jwtToken');
    if (!token) {
      this.props.history.push('/signin');
    }
  }
   /**
   * @description routes to /recipes if user is authenticated
   *
   * @method
   *
   * @memberOf AddRecipe
   *
   * @param {Object} nextProps
   *
   * @returns {void}
   *
   */
  componentWillReceiveProps(nextProps) {
    if (nextProps.recipeReducer.isCreated) {
      this.props.history.push('/recipes');
    }
  }


  handleChange(e) {
    e.preventDefault();
    this.setState({ errors: {} });
    this.setState({ [e.target.name]: e.target.value });
  }

  /**
   * @description get image input and add to state
   *
   * @method
   *
   * @memberOf AddRecipe
   *
   * @param {Event} e
   *
   * @returns {void}
   */
  handleImageChange(e) {
    e.preventDefault();
    const file = e.target.files[0];
    this.setState({ errors: {}, image: {} });
    this.setState({ image: file });
  }

  /**
   *
   * @description submits form
   *
   * @method
   *
   * @memberOf AddRecipe
   *
   * @param {Event} e
   *
   * @returns {void}
   *
   */
  handleOnsubmit(e) {
    e.preventDefault();
    const err = validateAddRecipe(this.state);
    if (err.isError) {
      return this.setState({ errors: err.errors });
    }

    this.setState({ disabled: true });
    const formData = new FormData();
    formData.append('file', this.state.image);
    formData.append('upload_preset', CLOUDINARY_UPLOAD_PRESET);

    axios({
      url: CLOUDINARY_URL,
      method: 'POST',
      data: formData,
    }).then((data) => {
      this.setState({ imageUrl: data.data.secure_url, disabled: false });
      const {
        title, description, ingredients, procedures, imageUrl,
      } = this.state;
      console.log(imageUrl, 'image');
      this.props.createRecipe(title, description, ingredients, procedures, imageUrl);
    }).catch(() => { 
      this.setState({ disabled: false });
      Materialize.toast("Error! Please try again", 4000, 'red');
    });
  }

  /**
   *
   * @description renders JSX element
   *
   * @method
   *
   * @memberOf SignUp
   *
   * @returns {void}
   *
   */
  render() {
    return (
      <div className="body grey lighten-5">
        <UserMenu />
        <div className="main">
          <div className="container">
            <div className="addrecipe-form z-depth-4">
              <div className="col l6 m8 s12 offset-l6 offset-m4">
                <h4 className="center">Add Recipe</h4>
                <hr />
              </div>
              <form onSubmit={this.handleOnsubmit}>
                <div className="row">
                  <div className="input-field col s12">
                    <input
                      id="title"
                      name="title"
                      type="text"
                      value={this.state.title}
                      onChange={this.handleChange}
                    />
                    <label htmlFor="title" className="active">Title</label>
                  </div>
                  <span className="right red-text">
                    {this.state.errors.titleError}
                  </span>
                </div>
                <div className="row">
                  <div className="input-field col s12">
                    <input
                      id="description"
                      name="description"
                      type="text"
                      value={this.state.description}
                      onChange={this.handleChange}
                    />
                    <label
                      htmlFor="description"
                      className="active"
                    >Description
                    </label>
                  </div>
                  <span className="right red-text">
                    {this.state.errors.descriptionError}
                  </span>
                </div>
                <div className="row">
                  <div className="input-field col s12">
                    <textarea
                      id="ingredients"
                      name="ingredients"
                      value={this.state.ingredients}
                      onChange={this.handleChange}
                      placeholder="enter ingredients seperated by semicolon(;)"
                    />
                  </div>
                  <span className="right red-text">
                    {this.state.errors.ingredientsError}
                  </span>
                </div>
                <div className="row">
                  <div className="input-field col s12">
                    <textarea
                      id="procedures"
                      name="procedures"
                      value={this.state.password}
                      onChange={this.handleChange}
                      placeholder="enter procedures seperated by semicolon(;)"
                    />
                  </div>
                  <span className="right red-text">
                    {this.state.errors.proceduresError}
                  </span>
                </div>
                <div className="row">
                  <div className="input-field col s12">
                    <i className="material-icons prefix">insert_photo</i>
                    <input
                      type="file"
                      name="profileImage"
                      onChange={e => this.handleImageChange(e)}
                    />
                  </div>
                  <span className="right red-text">
                    {this.state.errors.imageError}
                  </span>
                </div>
                <div className="row">
                  <button
                    className="btn right green"
                    type="submit"
                    name="action"
                  >Post
                  </button>
                </div>
                <hr />
                <div className="row center">
                  <div>
                    {this.state.disabled ?
                      <SmallPreloader />
                    :
                    ''
                    }
                  </div>
                </div>
              </form>
            </div>
          </div>
        </div>
        <Footer />
      </div>
    );
  }
}

AddRecipe.propTypes = {
  createRecipe: PropTypes.func.isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired
  }).isRequired,
  recipeReducer: PropTypes.shape({
    isCreated: PropTypes.bool.isRequired
  }).isRequired
};

const mapStateToProps = state => ({
  recipeReducer: state.recipeReducer,
});

export default connect(mapStateToProps, { createRecipe })(AddRecipe);

