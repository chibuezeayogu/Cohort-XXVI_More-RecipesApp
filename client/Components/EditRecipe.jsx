
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import axios from 'axios';
import { connect } from 'react-redux';
import { editRecipe, getRecipe } from '../action/actionCreators';
import SmallPreloader from './SmallPreloader.jsx';
import Footer from './Footer/Footer.jsx';
import UserMenu from './Header/UserMenu.jsx';
import { validateEditRecipeForm, validateImage } from '../util/validateInputs';
import imageToFormData from '../util/ImageUpload';


/**
 * @description AddRecipe component for user to create a new recipe
 *
 * @class
 *
 * @extends Component
 *
 */
class EditRecipe extends Component {
   /**
   * @description initialize state and binds functiom
   *
   * @constructor
   *
   * @memberOf EditRecipe
   *
   * @returns {void}
   *
   */
  constructor() {
    super();
    this.state = {
      title: '',
      description: '',
      ingredients: '',
      procedures: '',
      imageUrl:'',
      image: {},
      selectedImage: '',
      errors: {},
      disabled: false,
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleOnsubmit = this.handleOnsubmit.bind(this);
  }

   /**
   * @description routes to /recipes if user is authenticated
   *
   * @method
   *
   * @memberOf EditRecipe
   *
   * @returns {void}
   *
   */
  componentWillMount() {
    const token = localStorage.getItem('jwtToken');
    if (!token) {
      this.props.history.push('/signin');
    } else {
      const { id } = this.props.match.params;
      this.props.getRecipe(id);
    }
  }

  /**
   * @description checks if next recipes is fetched and disables is loading
   *
   * @method
   *
   * @memberOf SingleRecipe
   *
   * @param {Object} nextProps - nextProps object
   *
   * @returns {void}
   */
  componentWillReceiveProps(nextProps) {
    const { isFetched, isUpdated, recipes } = nextProps.userRecipeReducer;
    const { id } = this.props.match.params;
    if (isFetched) {
      const { id } = this.props.match.params;
     const index = recipes.findIndex(recipe => recipe.id === parseInt(id, 10));
      this.setState({ isLoading: false,
        title: recipes[index].title,
        description: recipes[index].description,
        ingredients: recipes[index].ingredients,
        procedures: recipes[index].procedures,
        imageUrl: recipes[index].imageUrl });
    } else if (isUpdated) {
      this.props.history.goBack();
    }
  }

  /**
   * @description get input and add to state
   *
   * @method
   *
   * @memberOf EditRecipe
   *
   * @param {Event} event - event object
   *
   * @returns {void}
   */
  handleChange(event) {
    event.preventDefault();
    this.setState({ errors: {} });
    this.setState({ [event.target.name]: event.target.value });
  }

  /**
   * @description get image input and add to state
   *
   * @method
   *
   * @memberOf EditRecipe
   *
   * @param {Event} event - event object
   *
   * @returns {void}
   */
  handleImageChange(event) {
    event.preventDefault();
    const file = event.target.files[0];
    this.setState({ errors: {}, image: {}, selectedImage: '' });
    this.setState({ image: file });
  
    const reader = new FileReader();
    reader.onload = () => {
        this.setState({ selectedImage: reader.result });
    }
    reader.readAsDataURL(file);
  }

  /**
   *
   * @description submits form
   *
   * @method
   *
   * @memberOf EditRecipe
   *
   * @param {Event} event - event object
   *
   * @returns {void}
   *
   */
  handleOnsubmit(event) {
    event.preventDefault();
    const err = validateEditRecipeForm(this.state);
    if (err.isError) {
      return this.setState({ errors: err.errors });
    }
    // console.log(Object.keys(this.state.image), 'with');
    // if(Object.keys(this.state.image)) {
    //   const uploadData = imageToFormData(this.state.image);
    //   delete axios.defaults.headers.common['Authorization'];
    //   axios(uploadData)
    //     .then((data) => {
    //     this.setState({ imageUrl: data.data.secure_url });
    //     console.log(this.state, 'with');
    //     this.props.editRecipe(this.state);
    //   }).catch((error) => { 
    //     this.setState({ disabled: false });
    //     Materialize.toast("Error! Please try again", 4000, 'red');
    //   });
    // } else {
    //   console.log(this.state, 'without');
    const { id } = this.props.match.params;
      this.props.editRecipe(id, this.state);
    // }
  }

  /**
   *
   * @description renders JSX element
   *
   * @method
   *
   * @memberOf SignUp
   *
   * @returns {void}
   *
   */
  render() {
    const { isFetched, recipes} = this.props.userRecipeReducer;
    
    if (this.state.isLoading) {
      return (
        <div className="body grey lighten-5">
            <UserMenu {...this.props} />
            <div className="main">
              <div className="container">
                <Preloader />
              </div>
            </div>
          <Footer />
          </div>
        )
      }
      
    if(isFetched === false && !recipes) {
      return (
        <div className="body grey lighten-5">
          <UserMenu {...this.props} />
          <div className="main">
            <div className="container">
              <h4 className="center-align" style={{ marginTop: 200 }}>
                No recipe found</h4>
            </div>
          </div>
        <Footer />
        </div>
      )
    }

    return (
      <div className="body grey lighten-5">
        <UserMenu />
        <div className="main">
          <div className="container">
            <div className="edit-recipe-form">
              <div className="col l6 m8 s12 offset-l6 offset-m4">
                <h4 className="center">Edit Recipe</h4>
                <hr />
              </div>
              <form onSubmit={this.handleOnsubmit}>
                <div className="row">
                  <div className="input-field col s12">
                    <input
                      id="title"
                      name="title"
                      type="text"
                      value={this.state.title}
                      onChange={this.handleChange}
                    />
                    <label htmlFor="title" className="active">Title</label>
                  </div>
                  <span className="right red-text error-margin">
                    {this.state.errors.titleError}
                  </span>
                </div>
                <div className="row">
                  <div className="input-field col s12">
                    <input
                      id="description"
                      name="description"
                      type="text"
                      value={this.state.description}
                      onChange={this.handleChange}
                    />
                    <label
                      htmlFor="description"
                      className="active"
                    >Description
                    </label>
                  </div>
                  <span className="right red-text error-margin">
                    {this.state.errors.descriptionError}
                  </span>
                </div>
                <div className="row">
                  <div className="input-field col s12">
                    <textarea
                      id="ingredients"
                      name="ingredients"
                      value={this.state.ingredients}
                      onChange={this.handleChange}
                      placeholder="enter ingredients seperated by semicolon(;)"
                    />
                  </div>
                  <span className="right red-text error-margin">
                    {this.state.errors.ingredientsError}
                  </span>
                </div>
                <div className="row">
                  <div className="input-field col s12">
                    <textarea
                      id="procedures"
                      name="procedures"
                      value={this.state.procedures}
                      onChange={this.handleChange}
                      placeholder="enter procedures seperated by semicolon(;)"
                    />
                  </div>
                  <span className="right red-text error-margin">
                    {this.state.errors.proceduresError}
                  </span>
                </div>
                <div className="row">
                  <div className="file-field input-field"
                    style={{ marginLeft: 10, marginRight: 10 }}>
                    <div className="btn green">
                      <span> 
                        <i 
                          className="material-icons"
                        >insert_photo
                        </i>
                      </span>
                      <input 
                        type="file" 
                        multiple 
                        onChange={event => this.handleImageChange(event)} />
                    </div>
                    <div className="file-path-wrapper">
                      <input 
                        className="file-path validate" 
                        type="text"
                        placeholder="Upload recipe image" />
                    </div>
                    <span className="right red-text error-margin">
                    {this.state.errors.imageError}
                  </span>
                  </div>
                  <div 
                    className="row s12 m4 l4 center-align" 
                    style={{ width: '100%'}}>
                    <img src={this.state.selectedImage||this.state.imageUrl} 
                      className="responsive-img center z-depth-2" 
                      style={{ maxHeight: 400, minHeight: 'auto', maxWidth: 400, minWidth: 'auto' }}/>
                  </div> 
                </div>
                <div className="row">
                  <button
                    className="btn right green"
                    type="submit"
                    name="action"
                    disabled={this.state.disabled}
                    style={{ marginRight: 10 }}
                  >Update
                  </button>
                </div>
                <hr />
                <div className="row center">
                  <div>
                    {this.state.disabled ?
                      <SmallPreloader />
                    :
                    ''
                    }
                  </div>
                </div>
              </form>
            </div>
          </div>
        </div>
        <Footer />
      </div>
    );
  }
}

EditRecipe.propTypes = {
  editRecipe: PropTypes.func.isRequired,
  getRecipe: PropTypes.func.isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired
  }).isRequired,
  userRecipeReducer: PropTypes.shape({
    isFetched: PropTypes.bool.isRequired
  }).isRequired
};

EditRecipe.contextTypes = {
  router: PropTypes.object.isRequired
};

const mapStateToProps = state => ({
  userRecipeReducer: state.userRecipeReducer,
});

export default connect(mapStateToProps, { editRecipe, getRecipe })(EditRecipe);

