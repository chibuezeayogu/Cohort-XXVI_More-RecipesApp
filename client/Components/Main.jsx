import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Home from './Home.jsx';
import Menu from './Hearder/Menu.jsx';
import Footer from './Footer/Footer.jsx';

/**
 *
 * @description Main component for user to signin
 *
 * @class
 *
 * @extends Component
 *
 */
class Main extends Component {

  /**
   * @description checks if a user is signed in
   *
   * @method
   *
   * @memberOf Main
   *
   * @returns {void}
   *
   */
  componentWillMount() {
    const token = localStorage.getItem('jwtToken');
    if (token) {
      this.props.history.push('/recipes');
    }
  }

  /**
   * @description initailize parallax
   *
   * @method
   *
   * @memberOf Main
   *
   * @returns {void}
   *
   */
  componentDidMount() {
    $('.parallax').parallax();
  }
/**
   *
   * @description renders JSX element
   *
   * @method
   *
   * @memberOf Mian
   *
   * @returns {void}
 */
  render() {
    return (
      <div className="body">
        <Menu />
        <Home />
        <Footer />
      </div>
    );
  }
}

Main.propTypes = {
  history: PropTypes.shape({
    push: PropTypes.func.isRequired
  }).isRequired
};

export default Main;
