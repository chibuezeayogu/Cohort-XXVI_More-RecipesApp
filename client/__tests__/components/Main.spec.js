import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import { Main } from '../../Components/Main';

const props = {
  userData: {
    isAuthenticated: true,
    currentUser: { id: 1, email: 'chibuezeayogu@hotmail.com' }
  },
  history: {
    push: jest.fn()
  },
};

describe('<Main />', () => {
  it('should render Main Component', () => {
    const wrapper = shallow(<Main {...props} />);
    wrapper.instance().componentDidMount();
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<Main />', () => {
  it('should call componentWillMount', () => {
    const componentWillMountSpy = jest
      .spyOn(Main.prototype, 'componentWillMount');
    const wrapper = shallow(<Main {...props} />);
    expect(componentWillMountSpy).toHaveBeenCalled();
  });
});

