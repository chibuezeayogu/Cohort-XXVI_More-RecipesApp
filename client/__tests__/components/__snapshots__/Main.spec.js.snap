// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render Main Component 1`] = `
ShallowWrapper {
  "length": 1,
  "node": <div
    className="body"
  >
    <withRouter(Menu) />
    <Home />
    <Footer />
  </div>,
  "nodes": Array [
    <div
      className="body"
    >
      <withRouter(Menu) />
      <Home />
      <Footer />
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Main
        history={
          Object {
            "push": [MockFunction] {
              "calls": Array [
                Array [
                  "/recipes",
                ],
              ],
            },
          }
        }
        userData={
          Object {
            "currentUser": Object {
              "email": "chibuezeayogu@hotmail.com",
              "id": 1,
            },
            "isAuthenticated": true,
          }
        }
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Main {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "history": Object {
            "push": [MockFunction] {
              "calls": Array [
                Array [
                  "/recipes",
                ],
              ],
            },
          },
          "userData": Object {
            "currentUser": Object {
              "email": "chibuezeayogu@hotmail.com",
              "id": 1,
            },
            "isAuthenticated": true,
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="body"
        >
          <withRouter(Menu) />
          <Home />
          <Footer />
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="body"
        >
          <withRouter(Menu) />
          <Home />
          <Footer />
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Main
    history={
      Object {
        "push": [MockFunction] {
          "calls": Array [
            Array [
              "/recipes",
            ],
          ],
        },
      }
    }
    userData={
      Object {
        "currentUser": Object {
          "email": "chibuezeayogu@hotmail.com",
          "id": 1,
        },
        "isAuthenticated": true,
      }
    }
  />,
}
`;

exports[`<Main /> should render the Main Component 1`] = `
ShallowWrapper {
  "length": 1,
  "node": <div
    className="body"
  >
    <withRouter(Menu) />
    <Home />
    <Footer />
  </div>,
  "nodes": Array [
    <div
      className="body"
    >
      <withRouter(Menu) />
      <Home />
      <Footer />
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Main
        history={
          Object {
            "push": [MockFunction] {
              "calls": Array [
                Array [
                  "/recipes",
                ],
              ],
            },
          }
        }
        userData={
          Object {
            "currentUser": Object {
              "email": "chibuezeayogu@hotmail.com",
              "id": 1,
            },
            "isAuthenticated": true,
          }
        }
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Main {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "history": Object {
            "push": [MockFunction] {
              "calls": Array [
                Array [
                  "/recipes",
                ],
              ],
            },
          },
          "userData": Object {
            "currentUser": Object {
              "email": "chibuezeayogu@hotmail.com",
              "id": 1,
            },
            "isAuthenticated": true,
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="body"
        >
          <withRouter(Menu) />
          <Home />
          <Footer />
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="body"
        >
          <withRouter(Menu) />
          <Home />
          <Footer />
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Main
    history={
      Object {
        "push": [MockFunction] {
          "calls": Array [
            Array [
              "/recipes",
            ],
          ],
        },
      }
    }
    userData={
      Object {
        "currentUser": Object {
          "email": "chibuezeayogu@hotmail.com",
          "id": 1,
        },
        "isAuthenticated": true,
      }
    }
  />,
}
`;
